File: ShippingMgr.Core/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ShippingMgr.Core.Models
{
    public class Order:BaseEntity
    {
        [ForeignKey("Customer")]
        public long CustomerId { get; set; }
        public Customer Customer { get; set; }
        public DateTime Date { get; set; }
        public string Note { get; set; }
        public ICollection<OrderItem> Items { get; set; }
        public Order()
        {
            Date = DateTime.Now;
            CreatedAt = DateTime.Now;
            UpdatedAt = DateTime.Now;
        }
    }
}

File: ShippingMgr.Api/ClientApp/src/app/models/order.ts
import { Customer } from "./customer";
import { OrderItem } from "./orderItem";

export class Order{
    id: number;
    customerId: number;
    customer: Customer;
    date:Date;
    note: string;
    createdAt: Date;
    updatedAt:Date;
    items: OrderItem[] = [];
    constructor(init?: Partial<Order>){
        this.items = [];
        Object.assign(this, init);
    }

    fill(init?:Partial<Order>){
        Object.assign(this, init);
        if (this.items == null)
            this.items = [];
    }
}